File: Assets/Smash Way/Scripts JS/Const/NonGravity.js
#pragma strict



var ForceAdd:boolean=false;
var ForceVel:Vector3;
var ForceRot:Vector3;

var Sound:AudioClip;
var ForceVelDoing:Vector3;
var ForceRotDoing:Vector3;
private var OneDoing:boolean=false;


function Start () {
if(ForceAdd){
gameObject.GetComponent.<Rigidbody>().isKinematic = false;
gameObject.GetComponent.<Rigidbody>().velocity = ForceVel;
gameObject.GetComponent.<Rigidbody>().angularVelocity = ForceRot;
ForceAdd=false;
}
}


function OnTouchDown(){
Doing();
}

function OnMouseDown(){
Doing();
}

function Doing(){
if(!OneDoing){
OneDoing=true;
gameObject.GetComponent.<Rigidbody>().isKinematic = false;
gameObject.GetComponent.<Rigidbody>().velocity =
gameObject.GetComponent.<Rigidbody>().velocity + ForceVelDoing;
gameObject.GetComponent.<Rigidbody>().angularVelocity = 
gameObject.GetComponent.<Rigidbody>().angularVelocity + ForceRotDoing;
if(Sound){gameObject.GetComponent.<AudioSource>().PlayOneShot(Sound);}
}
}

File: Assets/Smash Way/Scripts CS/Const/NonGravity.cs
using UnityEngine;
using System.Collections;

public class NonGravity : MonoBehaviour {

	public bool ForceAdd=false;
	public Vector3 ForceVel;
	public Vector3 ForceRot;

	public AudioClip Sound;
	public Vector3 ForceVelDoing;
	public Vector3 ForceRotDoing;
	private bool OneDoing = false;


	void Start () {
		if(ForceAdd){
			gameObject.GetComponent<Rigidbody>().isKinematic = false;
			gameObject.GetComponent<Rigidbody>().velocity = ForceVel;
			gameObject.GetComponent<Rigidbody>().angularVelocity = ForceRot;
			ForceAdd=false;
		}
	}


	public void OnTouchDown(){
		Doing();
	}

	public void OnMouseDown(){
		Doing();
	}

	void Doing(){
		if(!OneDoing){
			OneDoing=true;
			gameObject.GetComponent<Rigidbody>().isKinematic = false;
			gameObject.GetComponent<Rigidbody>().velocity =
			gameObject.GetComponent<Rigidbody>().velocity + ForceVelDoing;
			gameObject.GetComponent<Rigidbody>().angularVelocity = 
			gameObject.GetComponent<Rigidbody>().angularVelocity + ForceRotDoing;
			if(Sound){gameObject.GetComponent<AudioSource>().PlayOneShot(Sound);}
		}
	}
}
